Lib/fontTools/feaLib/parser.py

```diff
@@ -838,6 +839,35 @@ def parse_table_BASE_(self, table):
             elif self.cur_token_ == ";":
                 continue
 
+    def parse_table_OS_2_(self, table):
+        statements = table.statements
+        numbers = ("FSType", "TypoAscender", "TypoDescender", "TypoLineGap",
+                   "winAscent", "winDescent", "XHeight", "CapHeight",
+                   "WeightClass", "WidthClass", "LowerOpSize", "UpperOpSize")
+        ranges = ("UnicodeRange", "CodePageRange")
+        os2 = {}
+        while self.next_token_ != "}":
```

Comment:
in the unlikely event that the table definition is incomplete and lacks the ending brace, doesn't this while block goes into infinite loop? This would apply not just here, but also elsewhere in the parser module, every time we do `while self.next_token_ != ...`.
I wonder if there is a way to prevent this from happening. Maybe checking if `self.next_token_` is not None as well (that should be the case when there's nothing else to parse, e.g. at the end of the file)?

```python
while (self.next_token_ and self.next_token_ != "}"):
```